python-scripts/eeprom-util.py

```diff
@@ -0,0 +1,198 @@
+  for byte in eepCryptData:
+    i = (i + 1) % 256
+    j = (j + eepKeyObj.state[i]) % 256
+    eepKeyObj.state[i], eepKeyObj.state[j] = eepKeyObj.state[j], eepKeyObj.state[i]
+    eepData.append(byte ^ eepKeyObj.state[(eepKeyObj.state[i] + eepKeyObj.state[j]) % 256])
+
+  eepKeyObj.i = i
+  eepKeyObj.j = j
+  return eepData
+
+parser = optparse.OptionParser()
+parser.add_option('-f', action='store', dest='eepFileLoc')
+parser.add_option('-k', action='store', dest='keyFileLoc')
+(options, args) = parser.parse_args()
+
+if options.eepFileLoc:
+  eeprom = Eeprom(options.eepFileLoc)
+  print ('\033[1mEncrypted EEPROM Data\033[0m')
+  print ('EEPROM Hash:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.hash)))
+  print ('Encrypted Confounder:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.confounder)))
+  print ('Encrypted HDD Key:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.hddKey)))
+  print ('Encrypted Game Region Code:\t{}'.format(''.join(format(x, '02x') for x in eeprom.gameReg)))
+
+  print ('\n\033[1mIdentity EEPROM Data\033[0m')
+  print ('Identity Checksum:\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.identChk)))
+  print ('Serial Number:\t\t\t{}'.format(eeprom.serial.decode('ascii')))
+  print ('MAC Address:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.mac)))
+  print ('Online Key:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.onlineKey)))
+
+  # print NTSC / PAL instead of their respective codes
+  if ''.join(format(x, '02x') for x in eeprom.videoStd) == '00014000':
+    print ('Video Standard:\t\t\tNTSC')  
+  elif ''.join(format(x, '02x') for x in eeprom.videoStd) == '00038000':
+    print ('Video Standard:\t\t\tPAL')
+  else:
+    print ('Video Standard:\t\t\t{}'.format(''.join(format(x, '02x') for x in eeprom.videoStd)))
+
+if options.eepFileLoc and options.keyFileLoc:
+  # load eeprom key
+  eepKeyFile = open(options.keyFileLoc, 'rb')
+  eepKey = eepKeyFile.read()
+  
+  # generate hmac-sha1 hash of eeprom key
+  eepKeyHash = hmac.new(eepKey, eeprom.hash, sha1).digest()
+
+  # initialize EepKeyObj method object
+  eepKeyObj = EepKeyObj()
+  eepKeyPrepCrypt(eepKeyHash, eepKeyObj)
+  
+  # decrypt confounder 
+  decConf = eepKeyCrypt(eeprom.confounder, eepKeyObj)
+  
+  # assemble encrypted HDD key and game region code into a single bytearray
+  eepData = bytearray()
+  for byte in eeprom.hddKey:
+    eepData.append(byte)
+  for byte in eeprom.gameReg:
+    eepData.append(byte)
+
+  # decrypt HDD key and game region code
+  decData = eepKeyCrypt(eepData, eepKeyObj)
+
+  # assemble decrypted EEPROM data into a single bytearray
+  hashData = bytearray()
+  for byte in decConf:
+    hashData.append(byte)
+  for byte in decData:
+    hashData.append(byte)
+
+  # hash decrypted EEPROM data
+  eepHash = hmac.new(eepKey, hashData, sha1).digest()
+
+  # make sure hashes match
+  if eepHash == eeprom.hash:
+    print ('\n\033[1mThe decrypted and encrypted EEPROM hashes match!\033[0m')
```

Comment:
It's not unicode. If it was unicode it would change every character.

[This is probably a VT100 escape code](http://www.climagic.org/mirrors/VT100_Escape_Codes.html). I've worked on other projects which had hundreds of lines of code to handle different terminals. Windows in particular can be troublesome (I think they use a custom API instead of VT100).